apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/CarrierClaimTransformerV2.java

```diff
@@ -197,28 +197,51 @@ private static ExplanationOfBenefit transformClaim(
                 line.getPerformingPhysicianUpin());
       }
 
-      // Update the responsible flag
-      performing.ifPresent(
-          p -> {
-            p.setResponsible(true);
-
-            // PRVDR_SPCLTY => ExplanationOfBenefit.careTeam.qualification
-            p.setQualification(
+      if (performing.isPresent()) {
+        // Update the responsible flag
+        performing.get().setResponsible(true);
+        // PRVDR_SPCLTY => ExplanationOfBenefit.careTeam.qualification
+        performing
+            .get()
+            .setQualification(
                 TransformerUtilsV2.createCodeableConcept(
                     eob, CcwCodebookVariable.PRVDR_SPCLTY, line.getProviderSpecialityCode()));
 
-            // CARR_LINE_PRVDR_TYPE_CD => ExplanationOfBenefit.careTeam.extension
-            p.addExtension(
-                TransformerUtilsV2.createExtensionCoding(
-                    eob, CcwCodebookVariable.CARR_LINE_PRVDR_TYPE_CD, line.getProviderTypeCode()));
-
-            // PRTCPTNG_IND_CD => ExplanationOfBenefit.careTeam.extension
-            p.addExtension(
-                TransformerUtilsV2.createExtensionCoding(
-                    eob,
-                    CcwCodebookVariable.PRTCPTNG_IND_CD,
-                    line.getProviderParticipatingIndCode()));
-          });
+        boolean matchingExtension =
```

Comment:
IMO it would be clearer if this was extensionExists or performingHasMatchingExtension just to be a bit more readable; we're checking that we havent already added this extension to avoid duplicates, right?

Reply:
done